README.md

```diff
@@ -2,19 +2,134 @@
 
 ## Bring 'Code as Data' to market with enterprises
 
+### Enable engine integration with 3rd parties
+
+- [ ] gitbase-spark-connector SQL 3rd party interop (Spark Thrift Server) [P0][DP]
+
+### Increase number of algorithms we can run
+
+- [ ] Support for diff tree [P0][DP]
+- [ ] Support for diff [P0][DP]
+- [ ] Gitbase support for low-level git UDFs [P1][DP]
+- [ ] Research options for gitbase plugins [P1][DP]
+- [ ] Research options for UDFs (gitbase or gitbase-spark-connector) [P1][DP]
+
+### Increase variety of data we can analyze
+
+- [ ] Provide specs for external data sources [P1][DP][PRODUCT]
+- [ ] Provide specs to expose dependency analysis [P2][LA]
+
+### Analyze current performance limits and improve them
+
+- [ ] Parallelized queries inside of a repository [P1][DP]
+- [ ] Parallelized index creation [P1][DP]
+- [ ] Make performance dashboard available to product team [P1][DP][PRODUCT]
+- [ ] Provide specs for distributed version (gitbase-spark-connector) [P2][DP]
+
+### Make enterprise friendly
+
+- [ ] gitbase-web SSO and RBAC [P1][APPS][PRODUCT]
+- [ ] bblfsh-web SSO and RBAC [P1][APPS][PRODUCT]
+- [ ] document gitbase audit logs and RBAC [P1][DP]
+
+### Simplify usage
+
+- [ ] Gitbase to expose only reachable objects [P0][DP]
+- [ ] Provide installers [P2][APPS][PRODUCT]
+- [ ] Provide specs for new Engine architecture (stand-alone server + config file) [PRODUCT]
+
+### Improve stability
+
+- [ ] Ensure gitbase works when dataset and indexes do not match [P0][DP]
+- [ ] Provide a single, default, performant and stable regex implementation for gitbase [P2][DP]
+- [ ] Stabilize Babelfish in production (support large-scale use from gitbase) [P0][LA]
+- [ ] Add engine cli integration tests [P0][APPS]
+
+### Online updates
+
+- [ ] Gitbase to support online updates [P1][DP]
+
+### Release necessary language support for enterprise
+
+- [ ] C# driver [P0][LA]
+
+### Get bblfsh ready for wide language support of our use cases
+
+- [ ] Make UAST v2 the standard interface for Babelfish [P0][LA]
+- [ ] Compact diffs of UAST trees [P1][LA]
```

Comment:
One thing that isn't clear in this document are the dependencies between teams that arise from these objectives. (I'm adding this comment here because this one happens to affect LA, but there are others).

I suppose some of that dependency information will probably fall out of the product planning cycle, but could we have the OKR document include some record of "shared" OKRs? E.g., we could record multiple teams, and/or assign numbers and say "required for 1.2.1" or something?